### PR TITLE
feature: Task type warning for multiclass label containing integer

### PR DESCRIPTION
**Pull request by DanBasson**
Reference Issues/PRs
Fixes https://github.com/deepchecks/deepchecks/issues/1820

What does this implement/fix? Explain your changes.
This address the first section in the issue.
I've added a warning when inferring task type as multiclass if we found more than 10 unique values in the label and
the label contains only integers.

Any other comments?
i would appreciate your comments.
if i'm in the right direction, will move to the second section.

### CODE REVIEW

File: deepchecks/tabular/dataset.py [hunk 1]
```diff
@@ -535,9 +535,17 @@ def train_test_split(self: TDataset,
         return self.copy(train_df), self.copy(test_df)
 
     @staticmethod
-    def _infer_label_type(label_col: pd.Series):
+    def _infer_label_type(label_col: pd.Series, infer_task_type_from_label_nunique: int = 10):
```
Comment (nirhutnik) [inline deepchecks/tabular/dataset.py:538]: As this attribute is used for raising a warning only, there's not need for it to be a parameter of the function.

File: deepchecks/tabular/dataset.py [hunk 2]
```diff
@@ -535,9 +535,17 @@ def train_test_split(self: TDataset,
...
         if is_categorical(label_col, max_categorical_ratio=0.05):
             if label_col.nunique(dropna=True) > 2:
+                if infer_dtype(label_col) == 'integer' \
+                        and label_col.nunique() > infer_task_type_from_label_nunique:
+                    get_logger().warning(
+                        'Attributes such as "label_type" are not mandatory, but in a case of ordinal integers, '
```
Comment (nirhutnik) [inline deepchecks/tabular/dataset.py:544]: Would rephrase the warning here to be clearer on what happened, and how to fix it.
Suggesting something like:
"Integer label has many unique values. In this case, deepchecks automatically infers label to be multiclass. If your label is a regression label and not multiclass, please use "label_type='regression_label'" when initializing your Dataset."

**Comment by DanBasson** [comment]
Hi @nirhutnik
Do you know why am i getting this failures?

**Comment by nirhutnik** [comment]
@DanBasson yes, one was a small PR test that checks if the PR has a label. Solved that for you by adding one :)
The other test that fails, though it's not mandatory, is the coverages test that checks if this change causes less coverage. If you'll add a test for that warning you're adding, it would be ok

File: deepchecks/tabular/checks/model_evaluation/train_test_prediction_drift.py [hunk 3]
```diff
@@ -172,6 +173,13 @@ def run_logic(self, context: Context) -> CheckResult:
                 train_prediction = train_prediction[:, [1]]
                 test_prediction = test_prediction[:, [1]]
         else:
+            if context.task_type == TaskType.MULTICLASS:
+                get_logger().warning(
+                    'deepchecks infers label to be multiclass and auto plot class prediction drift. '
+                    'It is recommended to plot prediction probabilities instead, in order to '
+                    'produce meaningful results. To do so, add "drift_mode=\'proba\'" '
+                    'to TrainTestPredictionDrift check'
```
[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/tabular/checks/model_evaluation/train_test_prediction_drift.py:181]: I know that in #1820 there was a request for a validation of some sort, but this seems unnecessary, and I tried to discuss this with others and we can't currently find how this could cause confusion that the other warning you added doesn't cover.

File: deepchecks/tabular/dataset.py [hunk 4]
```diff
@@ -546,6 +546,13 @@ def train_test_split(self: TDataset,
     def _infer_label_type(label_col: pd.Series):
         if is_categorical(label_col, max_categorical_ratio=0.05):
             if label_col.nunique(dropna=True) > 2:
+                if infer_dtype(label_col) == 'integer' \
+                        and label_col.nunique() > 10:
```
Comment (nirhutnik) [inline deepchecks/tabular/dataset.py:550]: I think that 10 is too large. There's an issue here as some problems (discrete ordinal features) can actually be tackled both as regression and as multiclass (e.g. 1-5 stars in reviews).
So maybe >=5?

File: deepchecks/tabular/checks/model_evaluation/train_test_prediction_drift.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanBasson) [reply deepchecks/tabular/checks/model_evaluation/train_test_prediction_drift.py:181]: i agree, will change it

File: deepchecks/tabular/dataset.py (hunk 4 above)
Comment (DanBasson) [reply deepchecks/tabular/dataset.py:550]: sure

**Review by nirhutnik — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]

**Comment by ItayGabbay** [comment]
@DanBasson thanks for your contribution!
